src/lib/Types/Primitives.hs

```diff
@@ -73,7 +73,9 @@ data PrimCon e =
       | ConRef (PrimCon e)
       -- Misc hacks
       | ExplicitDict e e  -- Dict type, method. Used in prelude for `run_accum`.
-      | DictHole (AlwaysEqual SrcPosCtx) e -- Only used during type inference
+      -- Only used during type inference
+      | DictHole (AlwaysEqual SrcPosCtx) e
+      | HandlerHole (AlwaysEqual SrcPosCtx) e  -- e is HandlerDictTy
```

Comment:
In DictHoles we need the SrcPosCtx in case we need to report an error message about failing to find a valid instance. But type-checking should ensure that we can always resolve the handlers, so it should be unnecessary here, right?

Reply:
Removed in #1071